openedx/core/djangoapps/content/course_structures/api/v0/tests_api.py

```diff
@@ -0,0 +1,136 @@
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Disconnect course_published event.
+        """
+        # If we don't disconnect then tests are getting failed in test_crud.py
+        SignalHandler.course_published.disconnect(listen_for_course_publish)
+        super(CourseStructureApiTests, cls).tearDownClass()
+
+    def _expected_blocks(self, block_types=None, get_parent=False):
+        """
+        Construct expected blocks.
+        Arguments:
+            block_types (list): List of required block types. Possible values include sequential,
+                                vertical, html, problem, video, and discussion. The type can also be
+                                the name of a custom type of block used for the course.
+            get_parent (bool): If True then add child's parent location else parent is set to None
+        Returns:
+            dict: Information about required block types.
+        """
+        blocks = {}
+
+        def add_block(xblock):
+            """
+            Returns expected blocks dict.
+            """
+            children = xblock.get_children()
+
+            if block_types is None or xblock.category in block_types:
+
+                parent = None
+                if get_parent:
+                    item = xblock.get_parent()
+                    parent = unicode(item.location) if item is not None else None
+
+                blocks[unicode(xblock.location)] = {
+                    u'id': unicode(xblock.location),
+                    u'type': xblock.category,
+                    u'display_name': xblock.display_name,
+                    u'format': xblock.format,
+                    u'graded': xblock.graded,
+                    u'parent': parent,
+                    u'children': [unicode(child.location) for child in children]
+                }
+
+            for child in children:
+                add_block(child)
+
+        course = self.store.get_course(self.course.id, depth=None)
+        add_block(course)
+
+        return blocks
+
+    def test_course_structure_with_no_block_types(self):
+        """
+        Verify that course_structure returns info for entire course.
+        """
+        with mock.patch(self.MOCK_CACHE, cache.get_cache(backend='default')):
+            with self.assertNumQueries(3):
+                structure = course_structure(self.course.id)
+
+        expected = {
+            u'root': unicode(self.course.location),
+            u'blocks': self._expected_blocks()
+        }
+
+        self.assertDictEqual(structure, expected)
+
+        with mock.patch(self.MOCK_CACHE, cache.get_cache(backend='default')):
+            with self.assertNumQueries(2):
+                course_structure(self.course.id)
+
+    def test_course_structure_with_block_types(self):
+        """
+        Verify that course_structure returns info for required block_types only when specific block_types are requested.
```

Comment:
Must: I think you need additional tests here. For example, have a document with blocks like 'html', and 'video', but only request 'video'. Then assert only 'video' is returned. Another case, have a document wiht certain blocks ('foo') but request blocks that simply are not in the document ('bar'), ensure the return is what you expect (empty?).